lib/ClangImporter/ImportType.cpp

```diff
@@ -2731,8 +2749,6 @@ ParameterList *ClangImporter::Implementation::importFunctionParameterList(
     }
 
     bool knownNonNull = !nonNullArgs.empty() && nonNullArgs[index];
-    // Specialized templates need to match the args/result exactly.
-    knownNonNull |= clangDecl->isFunctionTemplateSpecialization();
```

Comment:
With this removed, you might have fixed some of the crashes for the commented-out cases in `template-instantiation-irgen.swift`, e.g.,

https://github.com/swiftlang/swift/blob/e79760bbeda0eff5a1973e8305d27880dbf3b1f1/test/Interop/Cxx/templates/template-instantiation-irgen.swift#L58-L59

Could you see if those are fixed by your patch?

Reply:
Ah great point, thanks! Let me uncomment some of these.

Reply:
(related: rdar://142850017)